### PR TITLE
Proof of concept: model "runs" and sends data to CODAP (PT-186719024)

### PR DESCRIPTION
**Pull request by lublagg**
This PR does the following:
- Adds a `name` property to each device which is updated if the user changes the device name input
- Creates types for the device "inputs" or variables
- Allows user to update name of single variable (note: not dealing with multiple variables at the moment)
- Sends data to CODAP and opens table

Note, since this is only a proof of concept, this PR does **NOT** do the following, which will need to be implemented in subsequent work:
- Does not update the attribute names in the CODAP table if the device names change
- Does not update device names if the attribute names in the CODAP table change
- Does not handle multiple devices having the same name 

Question:
- Should we rename "variables" to "inputs"? It seems more descriptive to me.

### CODE REVIEW

File: src/components/App.tsx [hunk 1]
```diff
@@ -139,6 +85,46 @@ export const App = () => {
     });
   };
 
+  const handleNameChange = (e: React.ChangeEvent<HTMLInputElement>, deviceId: Id) => {
+    setModel(draft => {
+      const columnIndex = draft.columns.findIndex(c => c.devices.find(d => d.id === deviceId));
```
Comment (dougmartin) [inline src/components/App.tsx:90]: I'm my upcoming "add arrows" PR I've added some utility functions to lookup the column and device - we can use those here once they are merged.

File: src/components/App.tsx [hunk 2]
```diff
@@ -139,6 +85,46 @@ export const App = () => {
     });
   };
 
+  const handleNameChange = (e: React.ChangeEvent<HTMLInputElement>, deviceId: Id) => {
```
Comment (dougmartin) [inline src/components/App.tsx:88]: I also added a `hooks` folder in the add arrows PR and I think we may want to add a `useModel` hook that we can move all these handlers into so we keep the app component shorter.  Again we can do that in a follow on PR after the arrows PR is merged.

**Review by dougmartin — APPROVED** [review]
Looks good :+1:

File: src/components/App.tsx (hunk 1 above)
Comment (lublagg) [reply src/components/App.tsx:90]: Sounds good.

**Pull request merged**